source/slang/slang-parameter-binding.cpp

```diff
@@ -3787,173 +3681,19 @@ struct CompleteBindingsVisitor : ComponentTypeVisitor
     }
 };
 
-/// A visitor used by `_completeBindings`.
-///
-/// This visitor is used to follow up after the `CompleteBindingsVisitor`
-/// any ensure that any "pending" data required by the parameters that
-/// got laid out now gets a location.
-///
-/// To make a concrete example:
-///
-///     Texture2D a;
-///     IThing    b;
-///     Texture2D c;
-///
-/// If these parameters were laid out with `b` specialized to a type
-/// that contains a single `Texture2D`, then the `CompleteBindingsVisitor`
-/// would visit `a`, `b`, and then `c` in order. It would give `a` the
-/// first register/binding available (say, `t0`). It would then make
-/// a note that due to specialization, `b`, needs a `t` register as well,
-/// but it *cannot* be allocated just yet, because doing so would change
-/// the location of `c`, so it is marked as "pending." Then `c` would
-/// be visited and get `t1`. As a result the registers given to `a`
-/// and `c` are independent of how `b` gets specialized.
```

Comment:
I'm glad to see all this code go, but I am also thankful to my past self for at least including comments like this that *try* to explain what all of this complexity is trying to achieve/support.